osquery/tests/integration/tables/process_envs.cpp

```diff
@@ -14,27 +14,55 @@
 
 #include <osquery/tests/integration/tables/helper.h>
 
+#include <algorithm>
+#include <stdlib.h>
+#include <sys/types.h>
+#include <unistd.h>
+
 namespace osquery {
 
-class processEnvs : public IntegrationTableTest {};
-
-TEST_F(processEnvs, test_sanity) {
-  // 1. Query data
-  // QueryData data = execute_query("select * from process_envs");
-  // 2. Check size before validation
-  // ASSERT_GE(data.size(), 0ul);
-  // ASSERT_EQ(data.size(), 1ul);
-  // ASSERT_EQ(data.size(), 0ul);
-  // 3. Build validation map
-  // See IntegrationTableTest.cpp for avaialbe flags
-  // Or use custom DataCheck object
-  // ValidatatioMap row_map = {
-  //      {"pid", IntType}
-  //      {"key", NormalType}
-  //      {"value", NormalType}
-  //}
-  // 4. Perform validation
-  // EXPECT_TRUE(validate_rows(data, row_map));
+class ProcessEnvs : public IntegrationTableTest {
+ public:
+
+  ProcessEnvs()
+    : env_name("OSQUERY_TEST_ENV_NAME")
+    , env_value("osquery_test_env_value")
+  {
+  }
+
+  void SetUp() override {
+    ::setenv(&(env_name.front()), &(env_value.front()), 1);
+  }
+
+  void TearDown() override {
+    ::unsetenv(&(env_name.front()));
```

Comment:
You don't need an external parenthesis to apply `&`. It could be just `unsetenv(&env_name.front());`

Reply:
True, I was using `putenv()` before and didn't even notice the difference once I changed to `setenv()`. Thanks.